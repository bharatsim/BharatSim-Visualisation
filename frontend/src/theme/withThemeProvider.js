import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
