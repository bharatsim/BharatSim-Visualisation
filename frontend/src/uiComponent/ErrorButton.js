import React from 'react';
import PropTypes from 'prop-types';

import { Button, fade } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { ChildrenPropTypes } from '../commanPropTypes';

const useErrorButtonStyle = makeStyles((theme) => ({
  text: {
    boxShadow: 'unset',
    color: theme.palette.error.dark,
    ...theme.typography.body1,
    '&:hover': {
      backgroundColor: fade(theme.palette.error.light, 0.2),
    },
    '&:focus': {
      backgroundColor: fade(theme.palette.error.light, 0.2),
    },
    '&$disabled': {
      color: theme.palette.text.disabled,
    },
  },
  disabled: {},
}));

function ErrorButton({ variant, size, onClick, classes, children }) {
  const buttonClasses = useErrorButtonStyle();
  return (
    <Button
      variant={variant}
      size={size}
      onClick={onClick}
      classes={{ ...buttonClasses, ...classes }}
    >
      {children}
    </Button>
  );
}

ErrorButton.defaultProps = {
  variant: 'text',
  size: 'medium',
  classes: {},
};

ErrorButton.propTypes = {
  variant: PropTypes.string,
  size: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.shape({}),
  children: ChildrenPropTypes.isRequired,
};

export default ErrorButton;
