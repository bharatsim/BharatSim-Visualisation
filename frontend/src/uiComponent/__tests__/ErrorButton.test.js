import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import withThemeProvider from '../../theme/withThemeProvider';
import ErrorButton from '../ErrorButton';

const ErrorButtonWithProvider = withThemeProvider(ErrorButton);

describe('ErrorButton', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ErrorButtonWithProvider onClick={jest.fn}>Test </ErrorButtonWithProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <ErrorButtonWithProvider onClick={onClick}>Test</ErrorButtonWithProvider>,
    );

    const button = getByText('Test').closest('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalled();
  });
});
