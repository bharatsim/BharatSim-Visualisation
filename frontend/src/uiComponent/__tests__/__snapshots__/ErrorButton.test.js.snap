// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorButton should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-text-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test 
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
